### PR TITLE
[Python] Type check for Categorical and Datetime relies on pandas public API

### LINKED ISSUE #235: Deprecations in pandas 0.19 means moving to the public API
`pandas.core.common.is_categorical_dtype` -> `pandas.api.types.is_categorical_dtype`

`pandas.core.common.is_datetime64_any_dtype` -> `pandas.api.types.is_datetime64_any_dtype`

### PR DESCRIPTION
**Pull request by guillett**
Fixes #235. Depends on #238.

Substitutions as per advised by the warning message.

### CODE REVIEW

File: python/feather/ext.pyx [hunk 1]
```diff
@@ -24,7 +24,7 @@ from cython.operator cimport dereference as deref
 from libfeather cimport *
 
 import pandas as pd
-import pandas.core.common as pdcom
+import pandas.api.types as pdapi
```
Comment (jreback) [inline python/feather/ext.pyx]: this is not back-compat with feather, unless the min pandas version is changed to 0.19.0 (not likely).

I might create a section in ``feather/compat.py`` where you can do something like

```
if LooseVersion(pandas.__version__) < '0.19.0'):
    pdapi = pandas.core.common
else:
    pdapi = pandas.api.dtypes
```

then ``from feather.compat import pdapi``

Comment (guillett) [inline python/feather/ext.pyx]: Very good point and thanks a lot for the comment (I am pretty new here).

I forced pandas to 0.18.0 [https://travis-ci.org/guillett/feather/builds/166268471](url) and it failed as expected.

I added your suggestions to a branch on my repository [https://github.com/guillett/feather/tree/type_check_backward_compat](url) and run Travis for pandas 0.18.0 [https://travis-ci.org/guillett/feather/builds/166265966](url) and 0.19.0 [https://travis-ci.org/guillett/feather/builds/166265267](url).

Should I push those changes to the branch I used in my pull request?
Thanks again.

**Comment by jreback** [comment]
yes u can push to this PR

further i would like to see an additional travis build that fixes pandas at 0.18.0 for compat

it could be in the 'optional' section (see how we do this for pandas builds)

**Comment by guillett** [comment]
The current master fails on python 2.7 with pandas 0.19.0 so I added the suggested travis builds in #238 that fixes the problem.

I rebased that branch on #238 to use the additional builds.

**Comment by wesm** [comment]
appears this was superseded by #242

**Pull request closed without merging**